### PR TITLE
Search: invalid queries error handling

### PR DESCRIPTION
**Pull request by FlorianCassayre**
* also displays a better component on any other search-related error (override was not registered correctly)
* closes inveniosoftware/invenio-app-ils#587

(previously it displayed a standard message provided by `react-searchkit`)

Screenshot from 2020-10-01 10-55-46

### CODE REVIEW

**Review by jrcastro2 — APPROVED** [review]

File: src/lib/modules/SearchControls/SearchError.js [hunk 1]
```diff
@@ -1,21 +1,48 @@
...
+  renderInvalidQueryError = () => {
+    return (
+      <Segment placeholder textAlign="center">
+        <Header icon>
+          <Icon name="question" />
+          Unrecognized search query.
+        </Header>
+        <Header.Subheader>
+          The query you entered was not recognized. Please refer to the
+          documentation for more information.
```
Comment (ntarocco) [inline src/lib/modules/SearchControls/SearchError.js:16]: Can we remove the part `refer to the documentation`? We still have to create such doc! :)
There is an issue to create advanced searches, but it is not yet done...

```suggestion
          title: Search query error.
          msg: The query is malformed or contains invalid characters.
```

File: src/lib/modules/SearchControls/SearchError.js [hunk 2]
```diff
@@ -1,21 +1,48 @@
...
+  renderOtherError = () => {
+    return (
+      <Segment placeholder textAlign="center">
+        <Header icon>
+          <Icon name="warning" />
+          Something went wrong while fetching results.
+        </Header>
+        <Header.Subheader>
+          Please try again later. If the problem persists, contact the technical
+          support.
```
Comment (ntarocco) [inline src/lib/modules/SearchControls/SearchError.js:31]: Can we add the contact? E-mail? It should be in the `invenioConfig.APP`. Double check other places how we display it.

File: src/lib/modules/SearchControls/SearchError.js (hunk 1 above)
Comment (FlorianCassayre) [reply src/lib/modules/SearchControls/SearchError.js:16]: OK, I was thinking about that too; will change, thanks.

**Comment by FlorianCassayre** [comment]
Done, I also factored some code since we now have several search "error" messages (empty result set, invalid query and other server errors).

**Review by ntarocco — APPROVED** [review]

**Pull request merged**